ad/movement/trusts.md

```diff
@@ -198,6 +207,50 @@ When doing NTLM authentications across trusts, the trusting domain's domain cont
+
+This is done by creating "Shadow Security Principals" in the Bastion Forest, which are mapped to SIDs for high-privileged groups in the User Forest, and then adding users from the Bastion Forest as members of the Shadow Security Principals.
+
+Shadow Principals reside in a special container `CN=Shadow Principal Configuration,CN=Services` under the Configuration container on the Bastion Forest.
+
+### Enumeration
+
+1. Using the ADModule, one can simply run Get-ADTrust and look for a trust which has ForestTransitive set to True and SIDFilteringQuarantined set to False - this means that SID Filtering is disabled. `Get-ADTrust -Filter {(ForestTransitive -eq $True) -and (SIDFilteringQuarantined -eq $False)}`
+2. If one wants to enumerate if their current forest is managed by a Bastion Forest, they can look for `ForestTransitive` set to `True` and `SIDFilteringForestAware` set to `True`. In this case, TrustAttributes is also a very good indicator. It is `0x00000400` (`1024` in decimal) for PAM/PIM trust. Simplifying it, it is `1096` for PAM + External Trust + Forest Transitive.
+3. To enumerate the shadow security principals, its members from the current (bastion) forest and privileges in the user/production forest, one can use the following command from the ActiveDirectory module: `Get-ADObject -SearchBase ("CN=Shadow Principal Configuration,CN=Services," + (Get-ADRootDSE).configurationNamingContext) -Filter * -Properties * | select Name,member,msDS-ShadowPrincipalSid | fl`
+The following properties are the most interesting ones:
+- `name` - Name of the shadow principal
+- `member` - Members from the bastion forest which are mapped to the shadow principal. In our example, it is the Domain Administrator of defensiveps.local.
+- `msDS-ShadowPrincipalSid` - The SID of the principal (user or group) in the user/prodcution forest whose privileges are assigned to the shadow security principal.
+### Abusing the shadow principals
+1. If an attacker compromises the user listed in "member" above they can use the shadow principals. With the privileges achieved using shadow principals above, they can access the user forest using RDP (explicit credentials of the bastion user required), WMI, PowerShell Remoting etc. Please note that if Kerberos AES Encryption is not enabled for the PAM trust, an attacker needs to add the machines of the existing forest in `WSMan TrustedHosts` and use the `-Authentication Negotiate` option with PowerShell remoting cmdlets.
+2. One can also use `SIDHistor`y injection using `Mimikatz` to abuse the PAM trust.
+3. Overall, auditors should look for a way to compromise the Bastion (Red) forest to easily gain access to all the managed Production Forests.
```

Comment:
This needs to be inserted in the Practice part of this page instead of having it as a subsection of the Bastion Forest. Creating a subsection called "Shadow Principals" in "Practice" would be relevant in my opinion

Reply:
Done. I have also rephrased the section to make it more digestible and easy to understand.